composition-js/src/merging/merge.ts

```diff
@@ -815,6 +819,95 @@ class Merger {
     return this.metadata(sourceIdx).isFieldShareable(field);
   }
 
+  private isOverride(_sourceIdx: number, field: FieldDefinition<any>): boolean {
+    return !!this.getOverrideDirective(field);
+  }
+
+  private getOverrideDirective(field: FieldDefinition<any>): Directive<FieldDefinition<any>> | undefined {
```

Comment:
I think there is actually subtlety here regarding `@override` on types due to the possible presence of type extensions (in the graphQL sense of the way). What I mean here is that you could have a subgraph that looks like this:
```graphql
type T {
  a: Int
  b: Int
}

extend type T @override(from: "X") {
  c: Int
  d: Int
}
```
And what we want for this is that the `@override` applies to `c` and `d` but not to `a` and `b`, but currently it would apply to all of them due to how we're dealing with extensions in `definitions.ts`. And to be clear, the handling of extensions is a tad ugly (but the rational is essentially that most code don't about them so I didn't wanted to make everything more complicated due to their presence, but the result is that when you do care about it, it's a tad weird).

Long story short, when looking at the type-level `@override` directive, the code should check if `directive.ofExtension() === field.ofExtension()` and ignore the directive (for that field) if that's not the case.

Note: I forgot to handle this properly for `@shareable`, which is why you may not have see it there. I'll create a separate PR to fix it.

Reply:
I really do not understand what ofExtension() is for. I'll try to dig in and see if it makes more sense to me, but I may have to ask you about this on Monday.